### PR TITLE
Provide better mappings for gg and G

### PR DESCRIPTION
**Pull request by kballard**
Update the `gg` and `G` mappings to support a count and to redraw all candidates if the count is past the current candidates list.

Also map `<C-Home>` and `<C-End>`.

### CODE REVIEW

**Comment by kballard** [comment]
Why was `gg` mapped to `gg0z.`? I preserved that behavior, but I don't understand the point. `gg` already jumps to the first non-blank character in the line, which is what `0z.` will do, and beyond that, `z.` tries to center the line in the window, but you can't center the first line. So the only difference between `gg` and `gg0z.` is if you provide a count to jump somewhere else, the latter will always center it, whereas the former will use the more intelligent default behavior for deciding if it should be centered.

**Comment by kballard** [comment]
BTW, this PR was motivated by the fact that I usually type `1G` instead of `gg` to jump to the top, and it bugged me that `1G` would always jump to the end.

File: autoload/unite/mappings.vim [hunk 1]
```diff
@@ -606,6 +610,29 @@ function! s:print_message_log() "{{{
...
 endfunction"}}}
+function! s:cursor_top() "{{{
+  let unite = unite#get_current_unite()
+  if v:count == 0
+    exe 'normal!' (unite.prompt_linenr == 1 ? '2' : '') . 'gg0z.'
+  else
+    if v:count > len(unite.current_candidates) + (unite.prompt_linenr == 1)
+      call unite#view#_redraw_all_candidates()
+    endif
+    exe 'normal!' v:count . 'gg0z.'
```
Comment (Shougo) [inline autoload/unite/mappings.vim]: Please use `:execute` instead of `exe`.

File: autoload/unite/mappings.vim [hunk 2]
```diff
@@ -606,6 +610,29 @@ function! s:print_message_log() "{{{
...
+function! s:cursor_bottom() "{{{
+  if v:count == 0
+    call unite#view#_redraw_all_candidates()
+    normal! G
+  else
+    let unite = unite#get_current_unite()
+    if v:count > len(unite.current_candidates) + (unite.prompt_linenr == 1)
+      call unite#view#_redraw_all_candidates()
+    endif
+    exe 'normal!' v:count . 'G'
```
Comment (Shougo) [inline autoload/unite/mappings.vim]: Please use `:execute`.

File: autoload/unite/mappings.vim (hunk 1 above)
Comment (kballard) [inline autoload/unite/mappings.vim]: Updated

File: autoload/unite/mappings.vim (hunk 2 above)
Comment (kballard) [inline autoload/unite/mappings.vim]: Updated

**Comment by Shougo** [comment]
Thanks.

**Pull request merged**